src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -3045,6 +3045,148 @@ public static void XmlMembersMapping_CompositeType()
         Assert.Equal(requestBodyValue.composite.StringValue, requestBodyActual.composite.StringValue);
     }
 
+#if ReflectionOnly
+    [ActiveIssue(18076)]
+#endif
+    [Fact]
+    public static void Xml_HiddenDerivedFieldTest()
+    {
+        var value = new DerivedClass { value = "on derived" };
+        var actual = SerializeAndDeserialize<BaseClass>(value,
+@"<?xml version=""1.0""?>
+<BaseClass xmlns:xsd=""http://www.w3.org/2001/XMLSchema"" xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xsi:type=""DerivedClass"">
+  <value>on derived</value>
+</BaseClass>");
+
+        Assert.NotNull(actual);
+        Assert.Null(actual.Value);
+        Assert.Null(actual.value);
+        Assert.Null(((DerivedClass)actual).Value);
+        Assert.Equal(((DerivedClass)actual).value, value.value);
+    }
+
+    [Fact]
+    public static void Xml_DefaultValueAttributeSetToNaNTest()
+    {
+        var value = new DefaultValuesSetToNaN();
+        var actual = SerializeAndDeserialize(value,
+@"<?xml version=""1.0""?>
+<DefaultValuesSetToNaN xmlns:xsd=""http://www.w3.org/2001/XMLSchema"" xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"">
+  <DoubleField>0</DoubleField>
+  <SingleField>0</SingleField>
+  <DoubleProp>0</DoubleProp>
+  <FloatProp>0</FloatProp>
+</DefaultValuesSetToNaN>");
+        Assert.NotNull(actual);
+        Assert.Equal(actual, value);
```

Comment:
Same. Can you please put the expected value in front of the actual value?

`Assert.Equal(expected, actual)`